app/src/main/java/edu/wisc/ece/pinpoint/pages/pins/PinCommentAdapter.java

```diff
@@ -0,0 +1,71 @@
+package edu.wisc.ece.pinpoint.pages.pins;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+import androidx.annotation.NonNull;
+import androidx.navigation.NavController;
+import androidx.recyclerview.widget.RecyclerView;
+import com.google.android.material.card.MaterialCardView;
+import java.util.ArrayList;
+import edu.wisc.ece.pinpoint.R;
+import edu.wisc.ece.pinpoint.data.Comment;
+import edu.wisc.ece.pinpoint.data.User;
+import edu.wisc.ece.pinpoint.utils.FirebaseDriver;
+
+public class PinCommentAdapter extends RecyclerView.Adapter<PinCommentAdapter.PinCommentViewHolder> {
+
+    private final FirebaseDriver firebase;
+    private final NavController navController;
+    private final ArrayList<Comment> comments;
+    private Context parentContext;
+
+    public PinCommentAdapter(ArrayList<Comment> comments, NavController navController) {
+        firebase = FirebaseDriver.getInstance();
+        this.navController = navController;
+        this.comments = comments;
+    }
+
+    @NonNull
+    @Override
+    public PinCommentAdapter.PinCommentViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.view_comment_item, parent, false);
+        parentContext = parent.getContext();
+        return new PinCommentAdapter.PinCommentViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull PinCommentAdapter.PinCommentViewHolder holder, int position) {
+        Comment comment = comments.get(position);
+        User user = FirebaseDriver.getInstance().getCachedUser(comment.getAuthorUID());
+        holder.content.setText(comment.getContent());
+        holder.timestamp.setText(comment.getTimestamp().toString());
+        holder.username.setText(user.getUsername());
+    }
```

Comment:
Can we make clicking on user's profile pic or username take you to their profile? Also we don't seem to ever load the user's profile pic or fetch the user if they aren't already cached. Finally, instead of displaying the `timestamp.toString()` use `FormatUtils.formattedDateTime()` to display the timestamp.